misc/config_davical.txt
        RewriteEngine On
        <IfModule mod_headers.c>
                Header unset Access-Control-Allow-Origin
                Header unset Access-Control-Allow-Methods
                Header unset Access-Control-Allow-Headers
                Header unset Access-Control-Allow-Credentials

                # If your setup is working and complete you can replace the "*" with CardDavMATE instalation
                #  origin (protocol/host/port) - see the examples below (for multiple origins use multiple headers)
                #Header always set Access-Control-Allow-Origin "http://www.server.com"
                #Header always set Access-Control-Allow-Origin "https://www.server.com"
                #Header always set Access-Control-Allow-Origin "https://www.server.com:8080"

                Header always set Access-Control-Allow-Origin "*"
                Header always set Access-Control-Allow-Methods "GET,POST,OPTIONS,PROPFIND,REPORT,PUT,MOVE,DELETE,LOCK,UNLOCK"
                Header always set Access-Control-Allow-Headers "User-Agent,Authorization,Content-type,Depth,If-match,If-None-Match,Lock-Token,Timeout,Destination,Overwrite,X-client,X-Requested-With"
                Header always set Access-Control-Allow-Credentials true

                RewriteCond %{HTTP:Access-Control-Request-Method} !^$
                RewriteCond %{REQUEST_METHOD} OPTIONS
                RewriteRule ^(.*)$ $1 [R=200,L]
        </IfModule>


changelog.txt
CardDavMATE Changelog

version 0.9.6 [2012-04-25]:
- note: if you use Davical with cross-domain setup see the modified Apache configuration in misc/config_davical.txt (solved problem with CardDAV-Sync from Marten Gajda)
- note: if you use MacOS X Lion calendarserver it is recommended to re-patch your installation (see misc/readme_lion.txt and the changelog below)
- configuration - changed default syncInterval from 15000 to 30000
- added support for automatic detection of crossDomain option in globalAccountSettings, globalNetworkCheckSettings and globalNetworkAccountSettings (if not set to boolean true/false it is automatically detected; default is null - autodetect)
- added IE9+ support (non cross-domain setup only /IE9 & jQuery limitation/; minor graphics glitches due to missing CSS support for "disabled" html elements)
- added support for prefix and suffix (vCard N attribute)
- added Czech localization (cs_CZ)
- added Danish localization (da_DK) - thanks Niels Bo Andersen
- fixed search with upper case letters not present in globalSearchTransformAlphabet
- fixed parsing of double quoted vCard param values (who use this?)
- fixed Andorra address fields
- fixed minor graphics glitch with update notification after logout and new login
- updated auth module to support multiple resources in response + updated auth/config.inc with modified templates
- updated misc/config_davical.txt (better handling of preflighted OPTION requests)
- updated misc/calendarserver_CardDavMATE.diff (support for cross-domain queries with X-Requested-With header /upcoming Safari 5.2/)
- updated default globalSearchTransformAlphabet in config.js
- updated CSS detection rule for iPad (new screen resolution for iPad 3)
- updated jQuery to 1.7.2
- updated jQuery-UI to 1.8.19
- replaced .attr('value'[, ...]) with .val([...])
- many other fixes

version 0.9.5 [2012-03-12]:
- note: this release contains new configuration option (always use the latest config.js)
- configuration - added globalSearchTransformAlphabet setting - used for transformation of non-ASCII characters to ASCII (for search support)
- added search support (search is performed on the addressbook values in the interface + category names)
- added language selector to login screen (use globalInterfaceLanguage to set the preselected language)
- added hungarian localization (hu_HU)
- scrollable resource list (only if needed)
- minor bug fixes

version 0.9.4 [2012-02-29]:
- fixed vCard parameter parsing (TYPE=X,Y now works correctly again)
- fixed date parsing for BDAY and X-ANNIVERSARY attributes (dates with missing '-' caused non-RFC vCard error)
- correct "\n" unescaping in vCard values if the previous character is '\'
- updated phone parameter definitions (iOS5 uses "TYPE=voice" in addition of other types)
- correct header removing from addressbook if contact is removed
- changed default value for undefined phone type from "main" to "cell" ("main" is Apple specific)
- added "TYPE=internet" into EMAIL parameters if it is missing (better non-RFC vCard compatibility)
- added "fax" phone type (in addition of "home fax" and "work fax")
- use of .prop() instead of .attr() to set 'readonly', 'disabled', 'selected', 'checked' and 'src' properties
- minor fixes for Slovak and German (thanks Thomas Scheel) localizations
- other minor fixes

version 0.9.3 [2012-02-17]:
- added german localization (de_DE) - thanks Marten Gajda
- fixed invalid switch to PROPFIND if REPORT/sync-collection is unsupported (+ related bugfixes)
- changed DOM operations to fix Firefox only bugs with 0.9.2 (Firefox or jQuery bug?)
- fixed "active editor -> logout -> login" problem (inactive resource/addressbook list)

version 0.9.2 [2012-02-13]:
- security fix: correct globalAccountSettings cleanup after logout
- note: this release contains new configuration options (always use the latest config.js)
- configuration - increased default timeout value for configuration account templates from 10 to 30 seconds
- configuration - changed default configuration for Davical - works also if Davical is installed into subdirectory (instead of domain root directory)
- configuration - added globalContactStoreFN option - support for custom FN formatting instead of fixed 'prefix,last,middle,first,suffix' (see config.js)
- configuration - added globalCompatibility option - customizations for 3rd party clients
- configuration - globalCompatibility: added anniversaryOutputFormat option - supported values: 'apple' (default) and 'other' (see config.js)
- configuration - added globalInterfaceLanguage option (see config.js and the changelog below)
- added localization support (new localizations are welcome!)
- added english localization (en_US)
- added slovak localization (sk_SK)
- added italian localization (it_IT) - thanks Luca Ferrario
- added support for Oracle, SabreDav and probably many other CardDav servers
- added support for CATEGORIES attribute with auto-suggest for categories in active collection (category separator: Enter)
- 100% performance improvement for PUT requests (add/edit/move/... operations) - get the modified Etag from PUT response header instead of additional REPORT requests (if the server support this feature, otherwise get the modified vCard with new Etag from the server)
- notable performance improvements for DOM operations (contact loading and manipulation)
- correct '\' character escaping in vCard values
- removed write support for X-EVOLUTION-* attributes (caused many problems and Evolution already support the X-* attributes without "-EVOLUTION" prefix)
- many other fixes

version 0.9.1 [2012-01-29]:
- note: this release contains new configuration options (always use the latest config.js)
- configuration - added lockTimeOut option into globalAccountSettings and globalNetworkCheckSettings - this option is used by the LOCK method and sets the lock timeout for the resource (default: 10000 miliseconds)

config.js
/*
CardDavMATE - CardDav Web Client
Copyright (C) 2011-2012 Jan Mate <jan.mate@inf-it.com>

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as
published by the Free Software Foundation, either version 3 of the
License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License for more details.

You should have received a copy of the GNU Affero General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

// globalAccountSettings must be an array (can be undefined if you use globalNetworkCheckSettings or globalNetworkAccountSettings)
//  the href value is a "principal URL" - the last character in href must be '/'
//    principal URL != collection URL -> the client automatically detects collections for each principal URL
//    PROPER principal URL looks like:
//      https://server.com:8443/principals/users/USER/
//      https://server.com:8443/caldav.php/USER/
//    INVALID principal URL looks like:
//      https://server.com:8443/principals/users/USER/addressbook/	<- url to addressbook collection
//      https://server.com:8443/caldav.php/USER/addressbook/		<- url to addressbook collection
//      https://server.com:8443/principals/users/USER				<- missing '/'
//      https://server.com:8443/caldav.php/USER						<- missing '/'
// the crossDomain sets jQuery's ajax crossDomain value (must be true if your CardDavMATE installation has not the same [protocol,hostname,port] as your CardDav server - by default null = autodetect /detected setting is shown in the console/)
// the withCredentials sets jQuery's ajax withCredentials value for cross domain queries (if true, Access-Control-Allow-Origin "*" is not allowed)
// the syncInterval sets how often (in miliseconds) to asynchronously sync the active collection on background (but only if the browser window has focus)
// the timeOut sets the timeout for jQuery .ajax call (in miliseconds)
// the lockTimeOut sets the LOCK Timeout value (in miliseconds)
//var globalAccountSettings=[{href: 'https://server1.com:8443/principals/users/USERNAME1/', crossDomain: null, withCredentials: false, userAuth: {userName: 'USERNAME1', userPassword: 'PASSWORD1'}, syncInterval: 30000, timeOut: 30000, lockTimeOut: 10000}, {href: 'https://server1.com:8443/principals/users/USERNAME2/', crossDomain: null, withCredentials: false, userAuth: {userName: 'USERNAME2', userPassword: 'PASSWORD2'}, syncInterval: 30000, timeOut: 30000, lockTimeOut: 10000}, {href: 'https://server2.com:8443/principals/users/USERNAME/', crossDomain: null, withCredentials: false, userAuth: {userName: 'USERNAME', userPassword: 'PASSWORD'}, syncInterval: 30000, timeOut: 30000, lockTimeOut: 10000}];

// if set, the client authenticates against the href URL (the last character in href must be '/') and if the authentication is successful it appends the USER + '/' to end of href and sets the userAuth: {userName: USER, userPassword: PASSWORD}
// then the client uses the modified globalNetworkCheckSettings in the same way as the globalAccountSettings
// this option ivokes a login screen and disallows access until successfull authentication
// the additionalResources array can contain additional resources (shared resources accessible by all users), for example: additionalResources: ['company','customers'] ... href values for these resources are created in the same way as described above for the USER
// see globalAccountSettings for more information
// Lion server example (http + https setup; see misc/readme_lion.txt for server setup):
//var globalNetworkCheckSettings={href: 'http://lion.server.com:8008/principals/users/', additionalResources: [], crossDomain: null, withCredentials: false, syncInterval: 30000, timeOut: 30000, lockTimeOut: 10000}
//var globalNetworkCheckSettings={href: 'https://lion.server.com:8443/principals/users/', additionalResources: [], crossDomain: null, withCredentials: false, syncInterval: 30000, timeOut: 30000, lockTimeOut: 10000}
// Davical example (for cross-domain setup see misc/config_davical.txt):
//var globalNetworkCheckSettings={href: 'http://davical.server.com:8080/caldav.php/', additionalResources: [], crossDomain: null, withCredentials: false, syncInterval: 30000, timeOut: 30000, lockTimeOut: 10000}
// Davical example (CardDavMATE installed into Davical subdirectory - works out of the box, no additional setup required):
var globalNetworkCheckSettings={href: location.protocol+'//'+location.hostname+(location.port ? ':'+location.port: '')+location.pathname.replace(RegExp('/+[^/]+/*$'),'')+'/caldav.php/', additionalResources: [], crossDomain: null, withCredentials: false, syncInterval: 30000, timeOut: 30000, lockTimeOut: 10000}

// if set, the configuration is loaded from the network (using HTTP auth) - the returned configuration XML settings are added
//  to globalAccountSettings ... it is possible to combine this option with the globalAccountSettings although it is not recommended
// this option ivokes a login screen and disallows access until the client get correct XML configuration file from the server
// the syncInterval is currently unused (the configuration XML is loaded only once)
// the timeOut sets the timeout for jQuery .ajax call (in miliseconds)
//var globalNetworkAccountSettings={href: 'https://www.config-server.com/auth/', crossDomain: null, withCredentials: false, syncInterval: 0, timeOut: 30000};

// default interface language - see localization.js
//  supported languages (note: value is case sensitive):
//   en_US (US English)
//   de_DE (German; thanks Marten Gajda and Thomas Scheel)
//   it_IT (Italian; thanks Luca Ferrario)
//   hu_HU (Hungarian)
//   sk_SK (Slovak)
var globalInterfaceLanguage='fr_FR';

// compatibility settings
//  anniversaryOutputFormat:
//  different clients use different (and incompatible) approach to store anniversary date in vCard
//   Apple stores this attribute as 'itemX.X-ABDATE;TYPE=pref:2000-01-01\r\nitemX.X-ABLabel:_$!<Anniversary>!$_\r\n'
//   other clients store this attribute as 'X-ANNIVERSARY:2000-01-01\r\n'
//  choose 'apple' or 'other' (lower case) for your 3rd party client compatibility (you can chose both: ['apple', 'other'] but it can cause many problems in the future, for example: duplicate anniversary dates, invalid/old anniversary date in your clients, and many others ...)
var globalCompatibility={anniversaryOutputFormat: ['apple']}

// JavaScript localeCompare() or custom alphabet for addressbook sorting
//  custom alphabet is used by default because the JavaScript localeCompare() not support collation and often returns "wrong" result
//var globalSortAlphabet=null;	// use localeCompare()
var globalSortAlphabet='0123456789AÁÄBCČDĎEÉFGHIÍJKLĹĽMNŇOÓÖŐÔPQRŔSŠTŤUÚÜŰVWXYÝZŽaáäbcčdďeéfghiíjklĺľmnňoóöőôpqrŕsšßtťuúüűvwxyýzž';	// use custom alphabet sorting
// search functionality character equivalence (transformation to ASCII: key = regex text, value = result character)
var globalSearchTransformAlphabet={'[ÀàÁáÂâÄäÆæÃãÅåĀā]': 'a', '[ÇçĆćČč]': 'c', '[Ďď]': 'd', '[ÈèÉéÊêËëĒēĖėĘęĚě]': 'e', '[ÎîÏïÍíĪīĮįÌì]': 'i', '[ŁłĽľĹĺ]': 'l', '[ÑñŃńŇň]': 'n', '[ÔôÖöÒòÓóŒœØøŌōÕõ]': 'o', '[Řř]': 'r', '[ẞßŚśŠš]': 's', '[Ťť]': 't', '[ÛûÜüÙùÚúŪūŰűŮů]': 'u', '[ÝýŸÿ]': 'y', '[ŽžŹźŻż]': 'z'};

// set the collection sorting, displaying and storing FN attribute into vCard - use a pair of values separated by a comma (spaces are allowed)
//  possible values:
//   last (equivalents: surname, lastname, family)
//   middle (equivalents: middlename)
//   first (equivalents: firstname, given)
//   prefix (no equivalents) -> unsupported in the editor, but if the contact includes a prefix, it is retained
//   suffix (no equivalents) -> unsupported in the editor, but if the contact includes a suffix, it is retained
var globalCollectionSort='last,middle,first';
var globalCollectionDisplay='last,middle,first';
var globalContactStoreFN='prefix,last,middle,first,suffix';	/* if undefined globalCollectionDisplay value is used */

// set the URI handlers for EMAIL, TEL and URL attributes (set to null or comment out to disable)
var globalUriHandlerTel='tel:';	// if 'tel' is not supported by system/browser, you can use 'callto' or 'skype'
var globalUriHandlerEmail='mailto:';
var globalUriHandlerUrl='http://';	// the value is used only if no URI handler is defined in the URL

// update notification will be shown only to users with login names defined in this array (for example: ['admin','peter'])
//  if undefined (or empty), update notifications will be shown to all users
var globalNewVersionNotifyUsers=[];

// set the datepicker format (see http://docs.jquery.com/UI/Datepicker/formatDate for valid values)
var globalDatepickerFormat='yy-mm-dd';
// default country for new address fields (must be defined in addressTypes variable - see common.js)
var globalDefaultAddressCountry='us';
// if there is no X-ABADR defined for the ADR attribute and the country name not matches any country name defined in the common.js the globalDefaultAddressCountry is used unless you define alternativne country names here
//  the country must refer to existing country defined in the common.js and the regex is any regex string which matches the given country (note: regex match is case insensitive)
var globalAddressCountryEquivalence=[{country: 'de', regex: '^\\W*Deutschland\\W*$'}, {country: 'sk', regex: '^\\W*Slovensko\\W*$'}];
// the countries listed here are shown at the top of the ADR country list (for example: ['de','sk'])
//  the country must refer to an existing country defined in the common.js
var globalAddressCountryFavorites=[];

// editor hide information message (success, error) after X miliseconds
var globalHideInfoMessageAfter=1800;

// show login names in resource header information?
var globalResourceHeaderShowLogin=true;

// asynchronously sync resources on background every X miliseconds (used for detection of collection changes in resources)
var globalSyncResourcesInterval=30000;
